### ISSUE TITLE
Cannot read json from a file buffer

### ISSUE THREAD

**Issue by illumination-k** [pola-rs/polars#2738]
#### What language are you using?

**Python**

#### What version of polars are you using?

`0.13.5`

#### What operating system are you using polars on?

WSL2 (Ubuntu)

```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.6 LTS
Release:        18.04
Codename:       bionic
```

#### What language version are you using
python 3.8.3

#### Describe your bug.

I cannot read json generated by `polars` itself if using a real file. I can read from buffer though.

#### What are the steps to reproduce the behavior?

```python
import io
import polars as pl

df = pl.DataFrame(
    {
        "bools": [False, True, False],
        "bools_nulls": [None, True, False],
        "int": [1, 2, 3],
        "int_nulls": [1, None, 3],
        "floats": [1.0, 2.0, 3.0],
        "floats_nulls": [1.0, None, 3.0],
        "strings": ["foo", "bar", "ham"],
        "strings_nulls": ["foo", None, "ham"],
    }
)


def file_json():
    try:
        f = "small_row_oriented.json"
        df.to_json(f, row_oriented=True)
        _ = pl.read_json(f)
    except RuntimeError as e:
        print(e)

    try:
        f = "small_json_lines.json"
        df.to_json(f, json_lines=True)
        _ = pl.read_json(f)
    except RuntimeError as e:
        print(e)

    try:
        f = "small.json"
        df.to_json(f)
        _ = pl.read_json(f)
    except RuntimeError as e:
        print(e)


def buf_json():
    f = io.BytesIO()
    df.to_json(f)
    f.seek(0)
    read_df = pl.read_json(f)
    assert df.frame_equal(read_df)


file_json()
buf_json()
```

#### What is the actual behavior?

```
Other("Error(\"expected value\", line: 1, column: 1)")
Other("Error(\"expected value\", line: 1, column: 1)")
Other("Error(\"expected value\", line: 1, column: 1)")
```

attaching generated json. I rename its ext to txt because json cannot upload in Github.

[small.txt](https://github.com/pola-rs/polars/files/8121989/small.txt)
[small_json_lines.txt](https://github.com/pola-rs/polars/files/8121992/small_json_lines.txt)
[small_row_oriented.txt](https://github.com/pola-rs/polars/files/8121994/small_row_oriented.txt)

#### What is the expected behavior?

We should read json generated by `polars` itself.

**Comment by universalmind303** [comment]
This is related to the inconsistencies mentioned in #2640 

so the documentation around `read_json` is actually a bit unclear. 

> source
    Path to a file or a file like object.

The python implementation can only take in a string or buffer of a json representation of the serde implementation. 
```py
>>> import polars as pl

>>> df = pl.DataFrame({"col_0": [1, 2, 3]})
>>> s = df.to_json()
>>> print(s)

{"columns":[{"name":"col_0","datatype":"Int64","values":[1,2,3]}]}

>>> df = pl.read_json(s)

```

It can't take in a path to a json file.

**Comment by illumination-k** [comment]
Ok, I miss it.

At least we should change the documentation of `py-polars`. Anyway, I close this and move to #https://github.com/pola-rs/polars/issues/2640

**Comment by ritchie46** [comment]
I did not realize we couldn't read from a file buffer. That's a bug we could and should fix easily.

**Comment by illumination-k** [comment]
I think the easy way to enable `read_json` from a file buffer is like the following. For API consistency, IMO this is a better way. But this implementation causes a big change because we cannot read json from str (Maybe this is now standard). What do you think about it?

```rust
#[staticmethod]
#[cfg(feature = "json")]
pub fn read_json(py_f: PyObject) -> PyResult<Self> {
    use std::io::{BufReader, Read};
    let mut file = BufReader::new(get_file_like(py_f, false)?);
    let mut json = String::new();
    file.read_to_string(&mut json)?;
    let df: DataFrame =
        serde_json::from_str(&json).map_err(|e| PyPolarsEr::Other(format!("{:?}", e)))?;
    Ok(df.into())
}
```

**Comment by ritchie46** [comment]
Closed by #2761
